Add testing on python 3.9 to CI

After upgrading min version python >= 3.8 there were some problems with running MPI backend on python 3.9. We should investigate that and fix if possible.

```
Run python unidist/cli -m pytest unidist/test/
mpiexec: Error: unknown option "-o"
Type 'mpiexec --help' for usage.
Traceback (most recent call last):
  File "/usr/share/miniconda/envs/unidist/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/share/miniconda/envs/unidist/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/runner/work/unidist/unidist/unidist/cli/__main__.py", line 122, in <module>
    main()
  File "/home/runner/work/unidist/unidist/unidist/cli/__main__.py", line 118, in main
    runner.run()
  File "/home/runner/work/unidist/unidist/unidist/cli/base/runner.py", line 57, in run
    subprocess.run(command, check=True)
  File "/usr/share/miniconda/envs/unidist/lib/python3.9/subprocess.py", line 528, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['mpiexec', '-hosts', '10.1.1.42:1,10.1.1.42:1,10.1.1.42:2,', '-n', '1', 'python', '-m', 'pytest', 'unidist/test/', ':', '-n', '1', '-wdir', '/tmp', 'python', '/home/runner/work/unidist/unidist/unidist/core/backends/mpi/core/monitor.py', ':', '-n', '2', '-wdir', '/tmp', 'python', '/home/runner/work/unidist/unidist/unidist/core/backends/mpi/core/worker.py']' returned non-zero exit status 1.
Error: Process completed with exit code 1.
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -56,7 +56,7 @@ jobs:
         shell: bash -l {0}
     strategy:
       matrix:
-        python-version: ["3.8"]
+        python-version: ["3.8", "3.9"]
         backend: ["ray", "mpi", "dask", "multiprocessing", "python"]
     env:
       UNIDIST_BACKEND: ${{matrix.backend}}
@@ -92,7 +92,7 @@ jobs:
         shell: bash -l {0}
     strategy:
       matrix:
-        python-version: ["3.8"]
+        python-version: ["3.8", "3.9"]
         backend: ["ray", "dask", "multiprocessing", "python"]
     env:
       UNIDIST_BACKEND: ${{matrix.backend}}

--- a/unidist/cli/backends/mpi/runner.py
+++ b/unidist/cli/backends/mpi/runner.py
@@ -75,7 +75,7 @@ class MPIRunner(BackendRunner):
         """
         unidist_root = get_unidist_root()
         workers_dir = "/tmp"
-        command = ["mpiexec", "-hosts"]
+        command = ["mpiexec", "-host"]
         hosts_str = f"{get_localhost_ip()}:1,{get_localhost_ip()}:1,"
         for host, n in zip(self.hosts, self.num_cpus):
             hosts_str += host + ":" + n + ","

